### DISCUSSION TITLE
Add support for ClickHouse as a database

### DISCUSSION THREAD

**Discussion by alexey-milovidov** [umami-software/umami#1201] [Ideas]
To allow analytics for high traffic websites.

**Comment by alexey-milovidov** [comment]
Plausible, PostHog, Splitbee, Piwik.pro and many others are already using ClickHouse.

**Comment by Vultik** [comment]
For how many visitors/days the support of ClickHouse is relevant ?
What are the current limitation in term of sessions using a PSQL db ?

**Comment by alexey-milovidov** [comment]
Starting from one million visitors total or around 10 000 / day * 100 days.

**Comment by asbrwl** [comment]
Would this be an option, or a change in the core requirements of Umami.
Currently I'm running Umami for 4 sites generating about 10-15k traffic per month on an ARM based server.
Effectively its much cheaper than an x64 system. When I last checked Clickhouse did not have ARM support.

I think by making it a core requirement, it might remove some of the deployment flexibility of Umami

**Reply by alexey-milovidov** [reply]
@asbrwl ClickHouse has the first-class ARM support since March 2022.

**Comment by Daniel15** [comment]
Duplicate of https://github.com/umami-software/umami/discussions/909
